lib/active_fedora/file.rb

```diff
@@ -63,7 +63,15 @@ def ldp_connection
     # If this file has a parent with ldp#contains, we know it is not new.
     # By tracking exists we prevent an unnecessary HEAD request.
     def new_record?
-      !@exists && ldp_source.new?
+      !@exists && ldp_new_record?
```

Comment:
Tracking `@destroyed` might be another way to do it.

There is no explicit place in the code where *we* call `==` but when I step into the `target.delete(record)` line it takes me directly to the `==` so I *suspect* Ruby is internally calling `==` when deleting an element in an array.